File: BEmuJava/bemu/src/main/java/com/bemu/BEmu/IntradayTickRequest/ElementIntradayTickString.java
//------------------------------------------------------------------------------
// <copyright project="BEmuJava" file="BEmu.IntradayTickRequest.ElementIntradayTickString.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bemu.BEmu.IntradayTickRequest;

import com.bemu.BEmu.Name;
import java.lang.StringBuilder;

public class ElementIntradayTickString extends ElementParent
{
    private final String _value;
    private final String _name;

    ElementIntradayTickString(String name, String value)
    {
        this._name = name;
        this._value = value;
    }
    
    public Name name()
    {
    	return new Name(this._name);
    }
    
    public int numValues()
    {
    	return 1;
    }
    
    public int numElements()
    {
    	return 0;
    }
    
    public boolean isComplexType()
    {
    	return false;
    }
    
    public boolean isArray()
    {
    	return false;
    }
    
    public boolean isNull()
    {
    	return false;
    }
    
    public boolean hasElement(String name)
    {
    	return false;
    }
    
    public boolean hasElement(String name, boolean excludeNullElements)
    {
    	return false;
    }
    
    protected StringBuilder prettyPrint(int tabIndent)
    {
    	return super.prettyPrintHelper(tabIndent, this._value);
    }
    
    public Object getValue()
    {
    	return this._value;
    }
    
    public Object getValue(int index) throws Exception
    {
    	if(index == 0)
    		return this.getValue();
    	else
    		return super.getValue();
    }
    
    public String getValueAsString()
    {
        return this._value;
    }

    public String getValueAsString(int index) throws Exception
    {
        if (index == 0)
            return this.getValueAsString();
        else
            return super.getValueAsString(index);
    }
}

File: BEmu_cpp/src/IntradayTickRequest/ElementIntradayTickString.cpp
//------------------------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/IntradayTickRequest/ElementIntradayTickString.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------------------------

#include "IntradayTickRequest/ElementIntradayTickString.h"
#include "BloombergTypes/Name.h"
#include "BloombergTypes/ElementPtr.h"

namespace BEmu
{
	namespace IntradayTickRequest
	{
		ElementIntradayTickString::ElementIntradayTickString(const std::string& name, const std::string& value)
		{
			this->_name = name;
			this->_value = value;
		}

		ElementIntradayTickString::~ElementIntradayTickString()
		{
		}

		Name ElementIntradayTickString::name() const
		{
			Name result(this->_name.c_str());
			return result;
		}
		
		bool ElementIntradayTickString::hasElement(const char* name, bool excludeNullElements) const
		{
			return false;
		}

		std::ostream& ElementIntradayTickString::print(std::ostream& stream, int level, int spacesPerLevel) const
		{
			this->prettyPrintHelper(stream, level, spacesPerLevel, this->_value);
			return stream;
		}

		const char * ElementIntradayTickString::getValueAsString(int index) const
		{
			if(index == 0)
			{
				return this->_value.c_str();
			}
			else
				throw elementPtrEx;
		}
	}
}